### ISSUE TITLE
Support duel `authentication_options` types `directory-service-authentication` and `certificate-authentication` on  `aws_ec2_client_vpn_endpoint` resource

### ISSUE THREAD

**Issue by tombmj** [terraform-providers/terraform-provider-aws#8723]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently `aws_ec2_client_vpn_endpoint` resource only supports `authentication_options` type `directory-service-authentication` **OR** `certificate-authentication`. This resource should support multiple authentication options  (**AND/OR**).  `aws_ec2_client_vpn_endpoint` schema currently defines  `authentication_options` MaxItems as 1.

### New or Affected Resource(s)

* aws_ec2_client_vpn_endpoint

### Potential Terraform Configuration

```hcl
resource "aws_ec2_client_vpn_endpoint" "vpn" {
  server_certificate_arn = "${var.server_certificate_arn}"
  client_cidr_block = "${var.client_cidr_block}"

  authentication_options {
    type = "certificate-authentication"
    root_certificate_chain_arn = "${var.root_certificate_chain_arn}"
  }

  authentication_options {
    type = "directory-service-authentication"
    active_directory_id = "${var.active_directory_id}"
  }
}
```

### References

* #0000

**Comment by marcus-bynder** [comment]
Voting for this; was surprised to learn that this provider doesn't allow you to specify both authentication options, as mutual TLS combined with password auth is definitely the most secure solution.  Without this ability, this provider isn't capable of supporting prod-ready VPN config, IMHO.

**Comment by bflad** [comment]
Support for this has been merged and will release with version 2.58.0 of the Terraform AWS Provider, later this week. Thanks to @davidsung for the implementation. 👍
